It looks like there's an error due to numpy outputting a warning (warnings are treated as errors in the JAX test suite):
```
        # warn on rank reduction, which indicates an ill conditioned matrix
        if rank != order and not full:
            msg = "Polyfit may be poorly conditioned"
>           warnings.warn(msg, RankWarning, stacklevel=4)
E           numpy.RankWarning: Polyfit may be poorly conditioned
```
The best way to deal with this is to use the `jtu.ignore_warning()` decorator for your `np_fun`; you can see examples of this elsewhere in `lax_numpy_test.py`.

If you want to test this locally, you can run the test case with more generated cases. For example:
```bash
JAX_NUM_GENERATED_CASES=50 pytest -n auto tests/lax_numpy_test.py -k testPolyfit
```

--- a/docs/jax.numpy.rst
+++ b/docs/jax.numpy.rst
@@ -294,6 +294,7 @@ Not every function in NumPy is implemented; contributions are welcome!
     poly
     polyadd
     polyder
+    polyfit
     polyint
     polymul
     polysub

--- a/jax/_src/numpy/lax_numpy.py
+++ b/jax/_src/numpy/lax_numpy.py
@@ -1368,6 +1368,76 @@ def gradient(f, *varargs, axis: Optional[Union[int, Tuple[int, ...]]] = None,
 def isrealobj(x):
   return not iscomplexobj(x)
 
+_POLYFIT_DOC = """\
+Unlike NumPy's implementation of polyfit, :py:func:`jax.numpy.polyfit` will not warn on rank reduction, which indicates an ill conditioned matrix
+Also, it works best on rcond <= 10e-3 values.
+"""
+@_wraps(np.polyfit, lax_description=_POLYFIT_DOC )
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+  _check_arraylike("polyfit", x, y)
+  deg = core.concrete_or_error(int, deg, "deg must be int")
+  order = deg + 1
+  # check arguments
+  if deg < 0:
+    raise ValueError("expected deg >= 0")
+  if x.ndim != 1:
+    raise TypeError("expected 1D vector for x")
+  if x.size == 0:
+    raise TypeError("expected non-empty vector for x")
+  if y.ndim < 1 or y.ndim > 2:
+    raise TypeError("expected 1D or 2D array for y")
+  if x.shape[0] != y.shape[0]:
+    raise TypeError("expected x and y to have same length")
+
+  # set rcond
+  if rcond is None:
+    rcond = len(x)*finfo(x.dtype).eps
+  rcond = core.concrete_or_error(float, rcond, "rcond must be float")
+  # set up least squares equation for powers of x
+  lhs = vander(x, order)
+  rhs = y
+
+  # apply weighting
+  if w is not None:
+    _check_arraylike("polyfit", w)
+    w, = _promote_dtypes_inexact(w)
+    if w.ndim != 1:
+      raise TypeError("expected a 1-d array for weights")
+    if w.shape[0] != y.shape[0]:
+      raise TypeError("expected w and y to have the same length")
+    lhs *= w[:, newaxis]
+    if rhs.ndim == 2:
+      rhs *= w[:, newaxis]
+    else:
+      rhs *= w
+
+  # scale lhs to improve condition number and solve
+  scale = sqrt((lhs*lhs).sum(axis=0))
+  lhs /= scale[newaxis,:]
+  from . import linalg
+  c, resids, rank, s = linalg.lstsq(lhs, rhs, rcond)
+  c = (c.T/scale).T  # broadcast scale coefficients
+
+  if full:
+    return c, resids, rank, s, rcond
+  elif cov:
+    Vbase = linalg.inv(dot(lhs.T, lhs))
+    Vbase /= outer(scale, scale)
+    if cov == "unscaled":
+      fac = 1
+    else:
+      if len(x) <= order:
+        raise ValueError("the number of data points must exceed order "
+                            "to scale the covariance matrix")
+      fac = resids / (len(x) - order)
+      fac = fac[0] #making np.array() of shape (1,) to int
+    if y.ndim == 1:
+      return c, Vbase * fac
+    else:
+      return c, Vbase[:,:, newaxis] * fac
+  else:
+    return c
+
 
 @_wraps(np.reshape, lax_description=_ARRAY_VIEW_DOC)
 def reshape(a, newshape, order="C"):

--- a/jax/numpy/__init__.py
+++ b/jax/numpy/__init__.py
@@ -51,7 +51,7 @@ from jax._src.numpy.lax_numpy import (
     nanmax, nanmean, nanmin, nanprod, nanstd, nansum, nanvar, ndarray, ndim,
     negative, newaxis, nextafter, nonzero, not_equal, number,
     object_, ogrid, ones, ones_like, outer, packbits, pad, percentile,
-    pi, piecewise, poly, polyadd, polyder, polyint, polymul, polysub, polyval, positive, power,
+    pi, piecewise, poly, polyadd, polyder, polyfit, polyint, polymul, polysub, polyval, positive, power,
     prod, product, promote_types, ptp, quantile,
     r_, rad2deg, radians, ravel, ravel_multi_index, real, reciprocal, remainder, repeat, reshape,
     resize, result_type, right_shift, rint, roll, rollaxis, rot90, round, round_, row_stack,

--- a/tests/lax_numpy_test.py
+++ b/tests/lax_numpy_test.py
@@ -1379,10 +1379,41 @@ class LaxBackedNumpyTests(jtu.JaxTestCase):
     tol = max(jtu.tolerance(lhs_dtype, tol_spec),
               jtu.tolerance(rhs_dtype, tol_spec))
     # TODO(phawkins): there are float32/float64 disagreements for some inputs.
-    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=False,
-                            tol=tol)
-    self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=False, atol=tol,
-                          rtol=tol)
+    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=False, tol=tol)
+    self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=False, atol=tol, rtol=tol)
+
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_{}_deg={}_rcond={}_full={}_w={}_cov={}".format(
+          jtu.format_shape_dtype_string(shape, dtype),
+          deg,
+          rcond,
+          full,
+          w,
+          cov),
+      "shape": shape, "dtype": dtype, "deg": deg,
+      "rcond": rcond, "full": full, "w":w, "cov":cov}
+      for dtype in [dt for dt in float_dtypes if dt not in [jnp.float16, jnp.bfloat16]]
+      for shape in [shape for shape in one_dim_array_shapes if shape != (1,)]
+      for deg in [1, 2, 3]
+      for rcond in [None, -1, 10e-3, 10e-5, 10e-10]
+      for full in [False, True]
+      for w in [False, True]
+      for cov in [False, True, "unscaled"]))
+  def testPolyfit(self, shape, dtype, deg, rcond, full, w, cov):
+    rng = jtu.rand_default(self.rng())
+    tol_spec = {np.float32: 1e-4, np.float64: 1e-13, np.complex64: 1e-5}
+    if jtu.device_under_test() == "tpu":
+      tol_spec[np.float32] = tol_spec[np.complex64] = 2e-1
+    tol = jtu.tolerance(dtype, tol_spec)
+    _w = lambda a: abs(a) if w else None
+    args_maker = lambda: [rng(shape, dtype), rng(shape, dtype), rng(shape, dtype)]
+    jnp_fun = lambda x, y, a: jnp.polyfit(x, y, deg=deg, rcond=rcond, full=full, w=_w(a), cov=cov)
+    np_fun = jtu.ignore_warning(
+      message="Polyfit may be poorly conditioned*")(lambda x, y, a: np.polyfit(x, y, deg=deg, rcond=rcond, full=full, w=_w(a), cov=cov))
+    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=False, tol=tol)
+    self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=False, atol=tol, rtol=tol)
+
 
   @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_{}_amin={}_amax={}".format(

